src/wallet/walletdb.cpp

```diff
@@ -909,7 +909,7 @@ DBErrors WalletBatch::LoadWallet(CWallet* pwallet)
     if (wss.fIsEncrypted && (last_client == 40000 || last_client == 50000))
         return DBErrors::NEED_REWRITE;
 
-    if (last_client < CLIENT_VERSION) // Update
+    if (!has_last_client || last_client < CLIENT_VERSION) // Update
```

Comment:
Any reason to keep this check? If yes, maybe rename `last_client` to `highest_client`?

Reply:
> Any reason to keep this check? 

The purpose of this process is store (first condition) or update (second condition) the version of the last client who opened the wallet db. Plus, we don't want to write to db if the value is already there.

> maybe rename last_client to highest_client?

hmm, I don't think that `highest_client` represents what this variable really is. I mean, the comment that is above the variable declaration is saying "// Last client version to open this wallet, was previously the file version number".

Which doesn't necessarily need to be the "highest" version.

Reply:
Yeah, the comment seems wrong as well. If you open with Bitcoin Core 23.0, it will write the 23.0 CLIENT_VERSION. If you then open with 22.0, it will not update because `last_client < CLIENT_VERSION` evaluates to false?

Reply:
oh yeah, that is also wrong. Updating.. :)

Reply:
Hmm, indeed, that is a misnomer. The comment and variable name should be changed to reflect what it actually does.

Reply:
Hmm, I was going into the opposite direction actually: update the client version value if it's different than the new version. So we always know (and can log) which soft version opened the wallet db last (something like this https://github.com/bitcoin/bitcoin/pull/25383/commits/c318211ddd48d44dd81dded553afeee3bc41c89e).